### ISSUE TITLE
mergeOptions does not update title component in Android

### ISSUE THREAD

**Issue by tautvilas** [wix/react-native-navigation#5377]
### Issue Description

If I set custom title component in android and later try to change it to another component with mergeOptions that does not work. On IOS it works fine.

###  Code example

```javascript
// Screen methods
componentDidMount() {
    Navigation.mergeOptions(this.props.componentId, {
      topBar: {
        title: {
          component: {
            name: 'component.Comp1',
          }
        }
      }
    });
  }

onInteractiveAction() {
    Navigation.mergeOptions(this.props.componentId, {
      topBar: {
        title: {
          component: {
            name: 'component.Comp2',
          }
        }
      }
    });
  }
```

If onInteractiveAction is triggered by some touch action in screen Comp2 will become visible in IOS navbar but not in Android

---
### Environment
* React Native Navigation version: 3.0.0-alpha.7
* React Native version: 0.60.4
* Platform(s) (iOS, Android, or both?): Android
* Device info: Pixel 2 android
---

Thanks for developing this amazing project :) RNN is the  number one navigation solution for react native apps!

**Comment by tautvilas** [comment]
I have identified the root cause of the problem. This if condition always evaluates to true:

https://github.com/wix/react-native-navigation/blob/master/lib/android/app/src/main/java/com/reactnativenavigation/presentation/StackPresenter.java#L418

The View component does not change even if the options arrive with new RN component. That means the top bar component is never updated. The same problem should also happen for backgroundComponent

If you want I can do a PR for the fix, but need guidance what could be the correct key for HashMap. I tried to fix by having key as RN component name (String), but this information is not accessible in isRendered method.

@guyca

**Comment by guyca** [comment]
@tautvilas Any chance you could PR a failing Detox test? I'll pick it up from there

**Comment by tautvilas** [comment]
I will try

**Comment by guyca** [comment]
Duaplicate of https://github.com/wix/react-native-navigation/issues/4625

**Comment by tautvilas** [comment]
Hey, was this resolved?

**Comment by gp3gp3gp3** [comment]
Any update on this?  This is still an issue as of version `3.2.0`

**Comment by tautvilas** [comment]
This is a patch for StackPresenter that fixes the problem

```patch
418,420c418,420
<             if (titleControllers.containsKey(component)) {
<                 topBar.setTitleComponent(titleControllers.get(component).getView());
<             } else {
---
>             //if (titleControllers.containsKey(component)) {
>             //    topBar.setTitleComponent(titleControllers.get(component).getView());
>             //} else {
426c426
<             }
---
>             //}
```

**Comment by vodanh109** [comment]
Hi, any update for this issue?

**Comment by juddey** [comment]
@vodanh109 Well, the patch still works :wink:

**Comment by guyca** [comment]
Hey guys, the fix is available in [6.4.0](https://github.com/wix/react-native-navigation/blob/master/CHANGELOG.md#640), sorry for taking so long to address this.

I understand this was mostly used to replace a component with the same component, only with different props. If all you need is to update props then you should be using [Navigation.updateProps](https://wix.github.io/react-native-navigation/docs/passing-data-to-components#updating-props). This way the component isn't recreated and the existing component's props are updated.
